tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/type-guard.bal

```diff
@@ -1748,6 +1748,43 @@ function yoo(SomeRecord|int|() aa) {
 
 }
 
+type R1 readonly & record {|int i;|}?;
+type R2 (readonly & record {|int i;|})?;
+type R record {|int i;|};
+
+function testR1(R1 r) returns boolean {
+    return r is record {|int i;|};
+}
+
+function testR2(R2 r) returns boolean {
+    return r is record {|int i;|};
+}
+
+function testR3(R1 r) {
+    if r is R {
+        R _ = r; // no error
+    } else {
+        panic error("Type of 'r' should be 'R'");
+    }
+}
+
+function testR4(R2 r) {
+    if r is R {
+        R _ = r; // no error
+    } else {
+        panic error("Type of 'r' should be 'R'");
+    }
+}
+
+function testTypeTestingInReadonlyRecord() {
+    assertTrue(testR1({i: 2}));
+    assertTrue(testR2({i: 2}));
+    assertFalse(testR1(()));
+    assertFalse((testR2(())));
+    testR3({i: 2});
+    testR4({i: 2});
```

Comment:
Better to assert the field or something instead of relying on non-panic behaviour.

Reply:
Changed